api/desecapi/models.py

```diff
@@ -281,6 +281,16 @@ def keys(self):
             self._keys = pdns.get_keys(self)
         return self._keys
 
+    @property
+    def touched(self):
+        try:
+            rrset_touched = max(updated for updated in self.rrset_set.values_list('touched', flat=True))
+            # If the domain has not been published yet, self.published is None and max() would fail
+            return rrset_touched if not self.published else max(rrset_touched, self.published)
+        except ValueError:
```

Comment:
Where does this ValueError come from? I understood comment above such that `max(None, some_timestamp)` would raise, but that is covered by the branching. Any other source of ValueErrors here?

Reply:
The `ValueError` comes from the `rrset_touched` assignment, if the `max()` argument iterable has no elements (i.e. there are no records). In this case, `max()` raises, we catch the exception and return `self.published` (which may or may not be `None` as well).